submissions/bdgenomics/scripts/1_integration.scala

```diff
@@ -0,0 +1,168 @@
+// parse the sample metadata
+val (samples, relations) = loadMetadata.time {
+  
+  // make hadoop paths for the metadata files
+  val relationsPath = new Path("%s/%s".format(dataDir, relatednessFile))
+  val samplePath = new Path("%s/%s".format(dataDir, populationsFile))
+
+  // get the filesystem for the sample metadata
+  val fs = relationsPath.getFileSystem(sc.hadoopConfiguration)
+
+  // open the two files
+  val relationsStream = fs.open(relationsPath)
+  val sampleStream = fs.open(relationsPath)
+
+  // extract the cryptic relatedness data
+  val parsedRelations = Source.fromInputStream(relationsStream)
+    .getLines
+    .drop(4) // first 4 lines are whitespace
+    .flatMap(line => {
+      // line is tab delimited:
+      // Population      Sample 1        Sample 2        Relationship    IBD0    IBD1    IBD2
+      val splitLine = line.split("\t")
+
+      if (splitLine.length != 7) {
+        Iterable.empty[Sample]
+      } else {
+        val population = splitLine(0)
+
+	// exclude samples in ASW that are siblings
+	// we will not filter them out
+	if (population == "ASW" &&
+	    splitLine(3).startsWith("Sibling")) {
+	  Iterable.empty[Sample]
+	} else {
+	  Iterable(Sample(splitLine(1), population),
+	    Sample(splitLine(2), population))
+	}
+      }
+    }).toSeq
+
+  // extract the population labels
+  val parsedPopulations = Source.fromInputStream(sampleStream)
+    .getLines
+    .flatMap(line => {
+      // line is tab delimited:
+      // SampleId Population ? SequencerPlatform
+      val splitLine = line.split("\t")
+      
+      if (splitLine.length > 2) {
+        Some(Sample(splitLine(0), splitLine(1)))
+      } else {
+        None
+      }
+    }).toSeq
+
+  // Scala.io.source is lazy --> force materialization
+  println("Have %d related samples across %d populations.".format(parsedRelations.size,
+    parsedPopulations.size))
+
+  (parsedPopulations, parsedRelations)
+}
+
+// load the rdd from disk
+val genotypes = sc.loadParquetGenotypes("%s/chr22.annotated.gt.adam".format(dataDir))
+
+// munge and broadcast the sample metadata
+val bcastSampleToPopulationMap = broadcastMetadata.time {
+  val relatedSamples = relations.map(_.sampleId).toSet
+  val sampleToPopulationMap = samples.filter(sample => relatedSamples(sample.sampleId))
+    .map(sample => (sample.sampleId, sample.population))
+    .toArray
+    .toMap
+ sc.broadcast(sampleToPopulationMap)
+}
+
+// are we caching the data? if so, cache it now
+val cachedGenotypes = cacheInputRdd.time {
+    
+  // cache the genotypes and force the genotypes to be materialized into
+  // memory by doing a count
+  val cachedGenotypes = genotypes.transform(_.cache)
+  println("Have %d genotypes.".format(cachedGenotypes.rdd.count))
+
+  cachedGenotypes
+}
+
+val variantsByPopulation = queryRdd.time {
+ cachedGenotypes.rdd
+  .flatMap(gt => {
+      bcastSampleToPopulationMap.value
+       .get(gt.getSampleId)
+       .map(population => (population, gt))
+    }).filter(kv => {
```

Comment:
I wonder if there would be utility in `filterBy` method(s) on VariantRDD/GenotypeRDD to make this easier to read?  The filter would need to happen before the flatMap to `(population, gt)` then.  Is there much performance difference between a compound filter like this with lots of expressions and a series of `filter` calls with smaller expressions?